Public/Src/Tools/SymbolDaemon/Tool.SymbolDaemonRunner.dsc

```diff
@@ -165,15 +168,55 @@ function addFiles(createResult: SymbolCreateResult, args: OperationArguments, fi
         uberArgs,
         ipcArgs => ipcArgs.merge<Transformer.IpcSendArguments>({
             messageBody: [
-                ...fileMessageBody,                    
+                ...fileMessageBody,
             ],
             lazilyMaterializedDependencies: [
-                ...files,                    
+                // note: symbolMetadataFile is not included in this list
+                ...files,
             ],
         })
     );
 }
 
+function indexSymbolFiles(files: File[]) : DerivedFile {
+    const symbolDataFileName = "symbol_data.txt";
+
+    Contract.requires(
+        files.length !== 0,
+        "The list of files cannot be empty"
+    );   
+
+    // get a directory where we are going to store the result of indexing
+    const outDir = Context.getNewOutputDirectory("symbol_indexing");
+
+    // the result of indexing
+    const outputPath = p`${outDir.path}/${symbolDataFileName}`;
+
+    const executeArgumetns = <Transformer.ExecuteArguments>{
+        tool: tool,
+        workingDirectory: outDir,
+        arguments: [
+            Cmd.argument("indexFiles"),
+            Cmd.options("--file ", files.map(f => Artifact.none(f))),
+            Cmd.options("--hash ", files.map(f => Artifact.vsoHash(f))),
+            Cmd.option("--symbolMetadata ", Artifact.output(outputPath)),
+        ],
+        unsafe: {
+            // SymStoreUtil opens some files (namely .pdb) with ReadWrite access,
+            // this causes DFAs and other issues.
+            untrackedPaths: files

```

Comment:
You're untracking the file you're indexing? That doesn't seem legit. Also, all should be correctly specified as inputs otherwise if this runs on a different machine in distribution they won't be materialized

Reply:
discussed offline

---
In reply to: [327696192](https://github.com/microsoft/BuildXL/pull/921#discussion_r327696192) [](ancestors = 327696192)